cipher/kdf.c

```diff
@@ -1184,31 +1180,84 @@ selftest_pbkdf2 (int extended, selftest_report_func_t report)
   const char *errtxt;
   int tvidx;
   int fail_fips = gcry_fips_request_failure("selftest_pbkdf2", "fail");
+  char buf[64];
 
   for (tvidx=0; tv[tvidx].desc; tvidx++)
     {
       what = tv[tvidx].desc;
       if (tv[tvidx].disabled)
         continue;
+
+      /* Skip test with shoter passphrase in FIPS mode.  */
+      if (fips_mode() && tv[tvidx].plen < 14)
```

Comment:
This is interesting that we move this out of check_one, I see why you did it, but now we are deviating even further from the upstream.

Reply:
This is onlt the functional tests, to make the test code "better" and to avoid it to log "SUCCESS" for tests that were merely skipped due to too short p.
I can undo this change if you want.